lib/qemu.c

```diff
@@ -736,7 +736,7 @@ make_uri (guestfs_h *g, const char *scheme, const char *user,
     break;
   }
 
-  return (char *) xmlSaveUri (&uri);
+  return xmlURIUnescapeString((char*)xmlSaveUri(&uri), -1, NULL);
```

Comment:
I've no idea what the plan for escaping here is, or how this new parameter can be safely used.  The description of xmlURIUnescapeString says:

> Unescaping routine, does not do validity checks ! Output is direct unsigned char translation of %XX values (no encoding)

which does not fill me with hope either.

This also leaks memory.